xclim/generic.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8 -*-
+# Note: stats.dist.shapes: comma separated names of shape parameters
+# The other parameters, common to all distribution, are loc and scale.
+
+import dask
+import xarray as xr
+import numpy as np
+
+
+def select_time(da, **indexer):
+    """Select entries according to a time period.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+      Input data.
+    **indexer : {dim: indexer, }, optional
+      Time attribute and values over which to subset the array. For example, use season='DJF' to select winter values,
+      month=1 to select January, or month=[6,7,8] to select summer months. If not indexer is given, all values are
+      considered.
+
+    Returns
+    -------
+    xr.DataArray
+      Selected input values.
+    """
+    if not indexer:
+        selected = da
+    else:
+        key, val = indexer.popitem()
+        time_att = getattr(da.time.dt, key)
+        selected = da.sel(time=time_att.isin(val)).dropna(dim='time')
+
+    return selected
+
+
```

Comment:
This is where I have spent most of my time reviewing.  I think I have my head wrapped around it and everything seems functional but I found the mixing of resample and seasonal subsetting using the **indexer in a single function somewhat confusing.   

I wonder if it would it be better to have users explicitly run 'select_time' to subset the data for a given season or month before using freq_analysis or other in a separate call.  This is how I would do it my and I suppose I still could as the indexer is optional but it took me sometime do understand that it does both ...

Reply:
Are there applications where there is a time subset that would combined with any other resample frequency other than YS?

Reply:
Yes, for DJF, you'd want YS-DEC. 

To clarify: 
- `select_time` is the function that only selects a time period. 
- `select_resample_op` applies an operation on a selected period at a given resampling frequency. 
- `frequency_analysis` is the convenience function wrapping these operations together:
  - rolling average
  - min/max operation
  - period selection
  - resampling
  - `fa` on the resulting array. 

So basically, you could do the array preparation on your own, then call `fa` on the result. Since there are a lot of streamflow indices to compute (for all seasons, min/max, different rolling averages) I thought it was best to provide that kitchen sink function.